aiservices/vertexai.go

```diff
@@ -0,0 +1,115 @@
+package aiservices
+
+import (
+	"context"
+
+	"cloud.google.com/go/auth/credentials"
+	"github.com/cloudwego/eino-ext/components/model/gemini"
+	"github.com/cloudwego/eino/components/model"
+	"github.com/pkg/errors"
+	"google.golang.org/genai"
+)
+
+const (
+	vertexAIFastModelId      = "gemini-2.5-flash"
+	vertexAIReasoningModelId = "gemini-2.5-pro"
+)
+
+type VertexAIModelConfig struct {
+	Project         string
+	Location        string
+	CredentialsFile string
+}
+
+type googleVertexAIFastModel struct {
+	baseModel model.ToolCallingChatModel
+}
+
+var _ Model = &googleVertexAIFastModel{}
+
+func NewGoogleVertexAIFastModel(ctx context.Context, config VertexAIModelConfig) (Model, error) {
```

Comment:
Why create different constructors? Have a single constructor (private) and just pass the predefined modelId based on our opinionated fast or reasoning model. Also Model Provider is the interface through which client codes should get a model. So no need to expose these.